qlib/workflow/record.py

```diff
@@ -0,0 +1,343 @@
+
+        Parameters
+        ----------
+        tags : dict
+            dictionary of tag_name: String -> value: String.
+
+        Returns
+        -------
+        None
+        """
+        raise NotImplementedError(f"Please implement the `log_tags` method.")
+
+    def delete_tag(self, key):
+        """
+        Delete a tag from a run.
+
+        Parameters
+        ----------
+        key : str
+            the name of the tag to be deleted.
+
+        Returns
+        -------
+        None
+        """
+        raise NotImplementedError(f"Please implement the `delete_tag` method.")
+    
+    def log_artifact(self, local_path, artifact_path=None):
+        """
+        Log a local file or directory as an artifact of the currently active run.
+
+        Parameters
+        ----------
+        local_path : str
+            path to the file to write.
+        artifact_path : str
+            the directory in `artifact_uri` to write to.
+
+        Returns
+        -------
+        None
+        """
+        raise NotImplementedError(f"Please implement the `log_artifact` method.")
+
+    def log_artifacts(self, local_dir, artifact_path=None):
+        """
+        Log all the contents of a local directory as artifacts of the run.
+
+        Parameters
+        ----------
+        local_dir : str
+            path to the directory of files to write.
+        artifact_path : str
+            the directory in `artifact_uri` to write to.
+
+        Returns
+        -------
+        None
+        """
+        raise NotImplementedError(f"Please implement the `log_artifacts` method.")
+
+    def get_artifact_uri(self, artifact_path=None):
+        """
+        Get the absolute URI of the specified artifact in the currently active run.
+
+        Parameters
+        ----------
+        artifact_path : str
+            the directory in `artifact_uri` to write to.
+
+        Returns
+        -------
+        An absolute URI referring to the specified artifact or currently active Recorder.
+        """
+        raise NotImplementedError(f"Please implement the `get_artifact_uri` method.")
+
+
+class MLflowRecorder(Recorder):
+    '''
+    Use mlflow to implement a Recorder.
+    '''
+    def start_run(self, run_id=None, experiment_id=None, 
+                    run_name=None, nested=False):
+        if run_id is None:
+            run_id = self.recorder_id
+        if experiment_id is None:
+            experiment_id = self.experiment_id
+        if run_name is None:
+            run_name = self.recorder_name
+        # start the run
+        run = mlflow.start_run(run_id, experiment_id, run_name, nested)
+        # save the run id and artifact_uri
+        self.recorder_id = run.info.run_id
+        self.artifact_uri = run.info.artifact_uri
+        # set up file manager for saving objects
+        if self.artifact_uri.startswith('file:/'):
+            self.fm = FileManager(Path(urllib.parse.urlparse(self.artifact_uri).path))

```

Comment:
Why we have to maintain a file manager? 
Can we support it only with mlflow and tempfile?

Reply:
More Comments for File Manager